base_local_planner/src/simple_trajectory_generator.cpp

```diff
@@ -115,26 +119,32 @@ void SimpleTrajectoryGenerator::initialise(
       }
       max_vel[1] = std::min(max_vel_y, vel[1] + acc_lim[1] * sim_period_);
       max_vel[2] = std::min(max_vel_th, vel[2] + acc_lim[2] * sim_period_);
+      max_vel[3] = std::min(max_vel_th_spin, vel[2] + acc_lim[2] * sim_period_);
 
       min_vel[0] = std::max(min_vel_x, vel[0] - acc_lim[0] * sim_period_);
       min_vel[1] = std::max(min_vel_y, vel[1] - acc_lim[1] * sim_period_);
       min_vel[2] = std::max(min_vel_th, vel[2] - acc_lim[2] * sim_period_);
+      min_vel[3] = std::max(min_vel_th_spin, vel[2] - acc_lim[2] * sim_period_);
     }
 
     Eigen::Vector3f vel_samp = Eigen::Vector3f::Zero();
     VelocityIterator x_it(min_vel[0], max_vel[0], vsamples[0]);
     VelocityIterator y_it(min_vel[1], max_vel[1], vsamples[1]);
-    VelocityIterator th_it(min_vel[2], max_vel[2], vsamples[2]);
     for(; !x_it.isFinished(); x_it++) {
       vel_samp[0] = x_it.getVelocity();
       for(; !y_it.isFinished(); y_it++) {
         vel_samp[1] = y_it.getVelocity();
+
+        // allow higher ang vels if lin vel is low
+        VelocityIterator th_it = hypot(vel_samp[0], vel_samp[1]) > 2 * limits->min_vel_trans ?
```

Comment:
Why `2 * limits->min_vel_trans`?

Reply:
Some random magic number christof suggested. It's just a threshold for deciding when to apply the new ang vel limits. Could also hardcode it to 0.05, 0.1, 0.2, etc... So why not `2 * limits->min_vel_trans` instead :shrug:

Reply:
S: The problem for me is that we cannot directly tune this value, since it is coupled with `min_vel_trans`
I would prefer to add another parameter for it

Reply:
That's true, but this is not something that would require much tuning -- it's just some arbitrary small threshold. Feels unnecessary to add a parameter for it.

Reply:
can you add a comment in the code about this value? (about the `2 * limits->min_vel_trans`)
when we see a formula like that, we usually think it is theory-based, not heuristics